packages/snap/src/core/clients/token-api-client/TokenApiClient.ts

```diff
@@ -65,33 +67,34 @@ export class TokenMetadataClient {
     const data = await response.json();
     assert(data, TokenMetadataResponseStruct);
 
-    return data;
+    return data.map((item) => ({
+      ...item,
+      assetId: item.assetId as TokenCaip19Id,
+    })) as TokenMetadata[];
```

Comment:
Changing the returned data here goes against the intention of the `assert` right above, which is here to ensure that we are about to return data in the correct shape.

I suggest we adapt `TokenMetadataResponseStruct` to include the field `assetId`, and then do this:
```suggestion
const data = await response.json();
const dataWithAssetId = data.map((item) => ({
      ...item,
      assetId: item.assetId as TokenCaip19Id,
    }))

assert(data, TokenMetadataResponseStruct); // Also performs type narrowing
return data // So we don't need to cast here
```